### PR TITLE
CBP 14: Create candidate model

### PR DESCRIPTION
**Pull request by pazuzu-ua**
Would appreciate your comments and suggestions regarding the fields, etc...

### CODE REVIEW

File: backend/apps/candidates/models.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        _("Firstname"),
+        max_length=50,
+        db_index=True,
+    )
+    surname = models.CharField(
+        _("Surname"),
+        max_length=50,
+        db_index=True,
+    )
+    date_of_birth = models.DateField(_("Birth date"), blank=True, null=True)
```
Comment (devoorhees) [inline backend/apps/candidates/models.py]: You can also add:
```
from datetime import date
from django.core.validators import MaxValueValidator

date_of_birth = models.DateField(
        verbose_name=_("Birth date"),
        validators=[MaxValueValidator(limit_value=date.today)],
        blank=True,
        null=True,
    )
```

File: backend/apps/candidates/models.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    def age(self) -> int:
+        """Return age of the candidate"""
+        if not self.date_of_birth:
+            return 0
+
+        from datetime import date
+
+        today = date.today()
+        month_day = (self.date_of_birth.month, self.date_of_birth.day)
+        return today.year - self.date_of_birth.year - ((today.month, today.day) < month_day)
```
Comment (devoorhees) [inline backend/apps/candidates/models.py:138]: Maybe this code is simpler?
`return int((today - self.date_of_birth).days / 365.25)`

**Review by devoorhees — APPROVED** [review]

File: backend/apps/candidates/models.py (hunk 2 above)
Comment (LV-coding) [reply backend/apps/candidates/models.py:138]: Its simpler, but the proposed method with "365.25" in the birthday in certain years may fail and give the wrong age.

image

File: backend/apps/candidates/models.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        blank=True,
+        db_index=True,
+    )
+    email = models.EmailField(
+        _("Email address"),
+        max_length=100,
+        blank=True,
+        db_index=True,
+    )
+    notes = models.TextField(
```
Comment (LV-coding) [inline backend/apps/candidates/models.py:106]: Maybe worth delete "blank=True" in phone_number? We always need the candidate's contacts.
Email  50/50.  Its also required, but the candidate may not submit it.

File: backend/apps/candidates/models.py [hunk 4]
```diff
@@ -37,7 +43,17 @@ class GenderChoices(models.TextChoices):
         MALE = "M", _("Male")
         FEMALE = "F", _("Female")
         OTHER = "O", _("Other")
-    
+
+    class EnglishLevelChoices(models.TextChoices):
+        """This class provides enum for levels of English."""
+        UNKNOWN = "A0", _("Unknown")
+        BEGINNER = "A1", _("Beginner")
+        ELEMENTARY = "A2", _("Elementary")
```
Comment (LV-coding) [inline backend/apps/candidates/models.py:56]: A0 its beginner, A1 its elementary, A2 its pre-intermediate

File: backend/apps/candidates/models.py (hunk 1 above)
Comment (pazuzu-ua) [reply backend/apps/candidates/models.py]: Ok, thank you!

File: backend/apps/candidates/models.py (hunk 4 above)
Comment (pazuzu-ua) [reply backend/apps/candidates/models.py:56]: According to CEFR there is no such thing as A0 =) I made it up, just out of blue. I used the official table of English levels, https://www.coe.int/en/web/common-european-framework-reference-languages/table-1-cefr-3.3-common-reference-levels-global-scale

File: backend/apps/candidates/models.py (hunk 3 above)
Comment (pazuzu-ua) [reply backend/apps/candidates/models.py:106]: There is no ''blank=True" in phone number, it is unique and required. Do you mean to get rid of it in email?

Comment (LV-coding) [reply backend/apps/candidates/models.py:106]: About phone number, I didn't see that there was a new commit that fixes "blank=True"). I think yes.

**Review by LV-coding — APPROVED** [review]

**Pull request merged**